### DISCUSSION TITLE
Upcoming models and your thoughts

### DISCUSSION THREAD

**Discussion by kahst** [Nachtzuster/BirdNET-Pi#422] [General]
Hi everyone, 

thank you so much for maintaining this fantastic project and making it one of the most important engagement and outreach tools for BirdNET.

We (the BirdNET-Team) are currently working on a new generation of models, and I guess now is a good time to reach out to you and ask for feedback, ideas or your wishlist. Just a few remarks on the model (V3.0) which might have implications for the codebase or the project in general:

- we'll likely feature all 10,000 bird species plus a decent (1,000-2,000) amount non-bird species
- we'll support variable input lengths (bye bye 3-second snippets)
- we'll optimize for all kinds of platforms, including Raspberry Pi (mostly through our "birdnet" Python package) and will likely also have different flavors/sizes of our models
- we'll have a development/preview version soonish
- and: we'll finally remove the "nc" part of the cc license (which we'll keep for now) so people can use the model even in commercial settings; instead models will come with terms of use (no harm to animals, no military use, etc ...)

Is there anything you need from us? Any features you are waiting for and we never delivered? Let us know, now is a good time!

I also still administrate the "birdnetpi.com" domain and can point it to a different site if you want or even transfer the domain to someone who'd like to take care of it. 

Thanks again for your dedication and work,
Stefan

**Comment by tphakala** [comment]
Hi Stefan, thank you for reaching out!

As a BirdNET-Go developer, my wish list is:

1. Please make classifier training data Open Source and available for download
2. If 1. is not possible, please make it possible to augment the noise/background class with custom training data

In real-time analysis scenarios, there are some bird species which have a very high occurrence of false positive detections due to man-made noise, and these could be eliminated significantly by having the ability to augment default classifiers' noise/background class with end users' ambient noise.

**Reply by kahst** [reply]
Yes, we'll publish the data we used for training - unless it is already published. We'll use XC, Macaulay Library, iNaturalist, observation.org and the Animal Sound Archive. We have one relatively large proprietary dataset that we'll polish and then upload to Zenodo.

Regarding your second question: We thought about ways to allow finetuning of the original weights instead of replacing the classification head. I think it is possible, but the training process is somewhat different from what we have implemented for now. The new model will definitely do a better job in noisy environments, but if you really want to adapt to a specific environment, you'd probably still have to train a custom classifier. The custom training process also allows using negative samples, i.e., samples that result in high-scoring false positives - if you have many of those, it might be worth considering custom classifier training.

**Comment by dacracot** [comment]
Hi Stefan,

I'm the developer of [BirdNET-BarChart](https://github.com/dacracot/BirdNET-BarChart) and the moderator of the [BirdNET-Analyzer subreddit](https://www.reddit.com/r/BirdNET_Analyzer/).

I would like to post your statement above to the subreddit to reach a larger audience, if that is alright with you, or perhaps you would like to engage that community yourself so you may get notified of any discussions.

As a developer of BirdNET-BarChart, I don't have any pressing enhancements that I want to suggest.  I do want to thank you and your team for a fantastic product and let you know of the many hours of pleasure I derive from working with it.

**Reply by kahst** [reply]
Yes, of course, thank you!

**Comment by tomtom215** [comment]
This is huge! 

One small enhancement I would love is that when running multiple microphones with birdnet, you are bound to have overlapping duplicate detections (not always exact same timestamp due to clock drift in streams over the network) - so a trusted de-duplication process would be very nice. This would allow users to add more mics without worry about any extra configs.

Thank you again for your on-going efforts for this community!

**Comment by alexbelgium** [comment]
Hello,

Thank you it is great to hear about this upcoming new model !

Beyond what was said, it would be interesting to tweak the "occurence" for specific species : currently night birds have very low occurrences levels that are probably due to biases in the eBird data. Having a way to tweak those would allow to re-add them without needing to reduce in an important manner the occurence threshold.

Also, I know that you have modified the Birdnet Analyzer to open the door to bats analysis, do you plan one day to also release a bats model ? Like @rdz-oss has implemented through his https://github.com/rdz-oss/BattyBirdNET-Pi

Best regards,
Alexandre

**Comment by jmtmp** [comment]
Many thanks for your work!
As far as I know, the model evaluates the probability of a bird's occurrence based on location. It would not be possible to work similarly with time, for example month/season and/or night/day. See also the previous idea for improving nocturnal bird identification.
Martin.

**Comment by Svardsten53** [comment]
There are now problems with human speech being included in recordings and for privacy reasons it would be good if this filtering could be improved. Thank you!

**Comment by sajidulhsajid-creator** [comment]
Much appreciate all the efforts you and the team putting into this project!

I am working on a project, which will have 3 remote birdnet-pi stations, which will be used to monitor an endangered species in Australia, called Mallefowl _(Leipoa Ocellata)_ .

I am currently running my pi's with a custom classifier, but I am not finding that accurate result so far. 

My wishlist is :

1. Having Malleefowl's identifiable.
2. Enabled GPS PPS timestamping for helping with accuracy. 

Thanks again!

**Comment by harrymnel** [comment]
Hi @kahst, how similar are the new models going to be to Google's new Perch 2.0? And will they have similar tools to train your own models? 

Thanks for all the efforts with BirdNet!!

**Comment by jbeale1** [comment]
We have a few Rhode Island Red chickens in a henhouse near my backyard bird listening station. A generic chicken-sounding croaking-like call was detected as "Common Raven" with 0.981 confidence. I understand that domestic chickens were not included in the training set, but wrong ID with confidence level this high seems unfortunate. I think I can't put sound files here but I enclose linear and log spectrograms and time-domain waveform plots as images, for two different calls (I have many more similar, also detected as raven). Any chance this miss-ID situation could be improved with some training tweaks?

$ pip show birdnet-analyzer
Name: birdnet_analyzer
Version: 2.4.0
Summary: BirdNET analyzer for scientific audio data processing and bird classification.

<img width="1500" height="450" alt="raven_waveform2" src="https://github.com/user-attachments/assets/fea1c40c-2ba4-4907-b452-7e2f1cbea8e8" />
<img width="1500" height="450" alt="raven_waveform" src="https://github.com/user-attachments/assets/5b3ea053-68ef-4831-9144-06feadae0479" />
<img width="1500" height="900" alt="raven_spectrogram2_log" src="https://github.com/user-attachments/assets/7be13374-4071-4a6b-944e-a14a4414ab33" />
<img width="944" height="613" alt="raven_spectrogram2_linear" src="https://github.com/user-attachments/assets/4647ee03-df47-4197-92b5-878be1061006" />
<img width="1500" height="900" alt="raven_spectrogram_log" src="https://github.com/user-attachments/assets/7400e2b2-d841-460f-99fa-26e839001b5f" />
<img width="944" height="613" alt="raven_spectrogram_linear" src="https://github.com/user-attachments/assets/3f9a02b1-2629-4c42-9171-435e822ea6fd" />
